packages/core/admin/admin/src/pages/SettingsPage/pages/Webhooks/EditView/components/HeadersInput/index.js

```diff
@@ -32,23 +32,22 @@ const HeadersInput = () => {
           name="headers"
           render={({ push, remove }) => (
             <Grid gap={4}>
-              {values.headers?.map((header, i) => (
-                // eslint-disable-next-line
-                <React.Fragment key={i}>
+              {values.headers.map((header, index) => (
+                <React.Fragment key={header.key}>
                   <GridItem col={6}>
                     <Field
                       as={Combobox}
-                      name={`headers.${i}.key`}
-                      aria-label={`row ${i + 1} key`}
+                      name={`headers.${index}.key`}
+                      aria-label={`row ${index + 1} key`}
                       label={formatMessage({
                         id: 'Settings.webhooks.key',
                         defaultMessage: 'Key',
                       })}
                       error={
-                        errors.headers?.[i]?.key &&
+                        errors.headers?.[index]?.key &&
                         formatMessage({
-                          id: errors.headers[i]?.key,
-                          defaultMessage: errors.headers[i]?.key,
+                          id: errors.headers[index].key,
+                          defaultMessage: errors.headers[index].key,
                         })
```

Comment:
I think you can render the error directly - no need to translate it:

```
errors.headers?.[index]?.key && errors.headers[index].key
```

Reply:
These formik errors give translation keys. e.g. without the translation the error would appear like this
<img width="406" alt="image" src="https://github.com/strapi/strapi/assets/48524071/efc3586c-483e-49b2-969a-449a6249429d">

Reply:
I think it would make sense to add the error messages to the yup schema instead, like we did here: https://github.com/strapi/strapi/blob/211b29142e1780b7b8a6829ed3313c640c545aa8/packages/core/admin/ee/admin/pages/SettingsPage/pages/ReviewWorkflows/utils/getWorkflowValidationSchema.js#L3-L34

What do you think?

Reply:
I see, makes sense. This is done in [`fcc9872` (#16800)](https://github.com/strapi/strapi/pull/16800/commits/fcc987225d2f25f49aa2d94407ab1b47af10c564)